packages/frontend/workspace/src/blob/cloud-blob-storage.ts

```diff
@@ -17,15 +18,15 @@ export const createCloudBlobStorage = (workspaceId: string): BlobStorage => {
         const suffix = predefinedStaticFiles.includes(key)
           ? `/static/${key}`
           : `/api/workspaces/${workspaceId}/blobs/${key}`;
+
         return fetchWithTraceReport(
           runtimeConfig.serverUrlPrefix + suffix
-        ).then(res => {
+        ).then(async res => {
           if (!res.ok) {
             // status not in the range 200-299
             return null;
           }
-          // todo: shall we add svg type here if it is missing?
-          return res.blob();
+          return bufferToBlob(await res.arrayBuffer());
```

Comment:
Do you think it's worth to add a smoke test for this case?

Reply:
Sure, let me add one

Reply:
Done